.github/workflows/workflow.yml

```diff
@@ -2,149 +2,128 @@ name: workflow
 
 on: [push, pull_request]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   lock:
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
     runs-on: "ubuntu-latest"
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - run: curl -sSL https://install.python-poetry.org | python3 -
       - run: poetry lock
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         with:
           path: poetry.lock
           key: ${{ github.sha }}-${{ matrix.python-version }}
-  pytest-dynalite:
+
+  pytest:
     needs: lock
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        implementation: ["dynalite", "dynamodb-local", "localstack"]
```

Comment:
You can set the matrix element to be a dictionary (or object as it's called in GHA) like so:
```yml
implementation: [{flavor: "other", name: "dynalite"}, {flavor: "other", name: "dynamodb-local"}, {flavor: "other", name: "localstack"}, {flavor: "scylla", name: "scylla"}]
```

Now you can refer them in env like so:
```yml
env:
  ...
  DYNAMODB_FLAVOR: ${{ matrix.implementation.flavor }}
  DYNAMODB_NAME: ${{ matrix.implementation.name }}
```

So now we can get rid of `pytest-scylla`.

Reply:
huh their documentation could really make this clearer...

Reply:
Couldn't agree more